slicedimage/backends/_caching.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import os
+
+from ._base import Backend
+
+
+class CachingBackend(Backend):
+    def __init__(self, cacheroot, authoritative_backend):
+        self._cacheroot = cacheroot
+        self._authoritative_backend = authoritative_backend
+
+    def read_file_handle_callable(self, name, checksum_sha1=None):
+        # TODO: (ttung) This is a very very primitive cache.  Should make this more robust with evictions and all that
```

Comment:
https://pypi.org/project/cachetools/ has some cool stuff that might work.

Reply:
cool, i'll check that out.